apps/sms/views/shared/js/utils.js

```diff
@@ -945,24 +945,33 @@
     },
 
     /**
-     * Returns Promise instance that is resolved instantly if document is
-     * visible (document.hidden === false), othewise it's resolved only once
-     * `visibilitychange` event is fired.
-     * @returns {Promise.<void>}
+     * Returns Promise instance that is resolved only once specified event is
+     * fired on the specified event target.
+     * @param {EventTarget} eventTarget Event target to listen event on.
+     * @param {String} eventName Name of the event to listen for.
+     * @returns {Promise.<*>} Promise resolved with event object if any.
      */
-    onceDocumentIsVisible() {
-      if (!document.hidden) {
-        return Promise.resolve();
-      }
-
+    onceEventIsFired(eventTarget, eventName) {
```

Comment:
I'd add optional ```timeout``` support here as well, but we don't need it anywhere yet, so didn't want to bring complexity here. The only downside here is that we'll leak ```'navigation-transition-end'``` event handler with such approach, what do you think?